hal/tests/TESTS/pin_names/arduino_uno/main.cpp

```diff
@@ -0,0 +1,300 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+#include "mbed.h"
+
+// #if !TARGET_FF_ARDUINO_UNO
+#if !(defined (TARGET_FF_ARDUINO) || (TARGET_FF_ARDUINO_UNO))
+#error [NOT_SUPPORTED] Test needs Arduino Uno form factor
+#else
+
+#if defined (TARGET_FF_ARDUINO)
+#warning ARDUINO form factor should not be used any more => use ARDUINO_UNO
+#endif
+
+using namespace utest::v1;
+
+template <PinName TestedPin>
+void GPIO_test()
+{
+    printf("Pin 0x%x\n", TestedPin);
+
+    TEST_SKIP_UNLESS_MESSAGE(TestedPin != (PinName)NC, "NC pin");
+    TEST_SKIP_UNLESS_MESSAGE(TestedPin != USBTX, "USBTX pin");
```

Comment:
Shouldn't these USBTX/USBRX checks be added to the generic pin tests?

Reply:
USBTX/USBRX are pins used by greentea, so they are implicitly tested